Gems/Pointcloud/Code/Source/Render/PointcloudFeatureProcessor.cpp

```diff
@@ -60,37 +61,84 @@ namespace Pointcloud
-        desc.m_bufferData = pcData.m_pointData.data();
+        const uint32_t elementCount = AZStd::max(static_cast<uint32_t>(cloudVertexData.size() + startIdx), pcData.m_vertices);
+        // new element count
+        const uint32_t bufferSize = elementCount * elementSize; // bytecount
 
-        pcData.m_cloudVertexBuffer = AZ::RPI::BufferSystemInterface::Get()->CreateBufferFromCommonPool(desc);
+        pcData.m_pointData.resize(bufferSize);
+        pcData.m_vertices = elementCount;
+        memcpy(pcData.m_pointData.data() + startIdx * elementSize, cloudVertexData.data(), cloudVertexData.size() * elementSize);
 
+        if (pcData.m_cloudVertexBuffer)
+        {
+            pcData.m_cloudVertexBuffer->Resize(bufferSize);
+            pcData.m_cloudVertexBuffer->UpdateData(pcData.m_pointData.data(), bufferSize, startIdx * elementSize);
+        }
+        else
+        {
+            AZ::RPI::CommonBufferDescriptor desc;
+            desc.m_poolType = AZ::RPI::CommonBufferPoolType::ReadWrite;
+            desc.m_bufferName = AZStd::string::format("PointcloudFeatureProcessor, %d", pcData.m_index);
+            desc.m_byteCount = bufferSize;
+            desc.m_elementSize = elementSize;
+            desc.m_bufferData = pcData.m_pointData.data();
+
+            pcData.m_cloudVertexBuffer = AZ::RPI::BufferSystemInterface::Get()->CreateBufferFromCommonPool(desc);
+        }
         pcData.m_meshStreamBufferViews.front() =
             AZ::RHI::StreamBufferView(*pcData.m_cloudVertexBuffer->GetRHIBuffer(), 0, bufferSize, elementSize);
 
-        pcData.m_drawSrg =
-            AZ::RPI::ShaderResourceGroup::Create(m_shader->GetAsset(), m_shader->GetSupervariantIndex(), m_drawSrgLayout->GetName());
-        m_pointSizeIndex.Reset();
-        m_modelMatrixIndex.Reset();
+        if (!pcData.m_drawSrg)
+        {
+            pcData.m_drawSrg =
+                AZ::RPI::ShaderResourceGroup::Create(m_shader->GetAsset(), m_shader->GetSupervariantIndex(), m_drawSrgLayout->GetName());
+            m_pointSizeIndex.Reset();
+            m_modelMatrixIndex.Reset();
+        }
 
         UpdateDrawPacket();
-        return pcData.m_index;
+    }
+
+    PointcloudFeatureProcessorInterface::PointcloudHandle PointcloudFeatureProcessor::AcquirePointcloudFromAsset(
+        AZ::Data::Asset<PointcloudAsset> pointcloudAsset)
+    {
+        m_currentPointcloudDataIndex++;
+        auto& pcData = m_pointcloudData[m_currentPointcloudDataIndex];
+        pcData.m_index = m_currentPointcloudDataIndex;
+        m_pointcloudAssets[pointcloudAsset.GetId()] = m_currentPointcloudDataIndex;
+        pcData.m_assetId = pointcloudAsset.GetId();
+        AZ::Data::AssetBus::MultiHandler::BusConnect(pointcloudAsset.GetId());
+        if (!(pointcloudAsset.IsLoading() || pointcloudAsset.IsReady()))
+        {
+            pointcloudAsset.QueueLoad();
+        }
+        if (pointcloudAsset.IsReady())
+        {
+            OnAssetReady(pointcloudAsset);
+        }
```

Comment:
genuine question: what's the benefit of OnAssetReady at this point?

Reply:
I'm not an expert but from what I understand this asset might have been loaded by some other component then OnAssetReady will not fire by itself